server/apps/publish/formatters/aap_ipnews_formatter.py

```diff
@@ -122,3 +174,110 @@ def format(self, article, subscriber):
 
     def can_format(self, format_type, article):
         return format_type == 'AAP IPNEWS' and article['type'] in ['text', 'preformatted']
+
+    def _set_selector_codes(self, article, subscriber_name, odbc_item, category):
+        desk_id = article.get('task', {}).get('desk', None)
+        if desk_id:
+            desk = superdesk.get_resource_service('desks').find_one(req=None, _id=desk_id)
+            if desk['name'] == 'NDX':
+                self._set_ndx_selector_codes(article, subscriber_name, odbc_item, category)
+            elif desk['name'] == 'SDX':
+                self._set_sdx_selector_codes(article, subscriber_name, odbc_item, category)
+            elif desk['name'] == 'FDX':
+                self._set_fdx_selector_codes(article, subscriber_name, odbc_item, category)
```

Comment:
This `if..elif` logic feels repetitive, it might make sense to make it more generic, for example:

```py
handlers = {
    'NDX': self._set_ndx_selector_codes,
    'SDX': self._set_sdx_selector_codes,
    'FDX': self._set_fdx_selector_codes,
}
...
handler = handlers.get(desk['name'],  lambda *args: None)
handler(article, subscriber_name, odbc_item, category)
```
Although in this particular example the gains are small (if any), since we only have three cases to check. But I would definitely consider it with 4+ cases or if it is likely that more selector codes are added.

Just a suggestion though, no need to change it as is.